google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -227,8 +230,11 @@ public CloudSdk createSdk(
         .appCommandCredentialFile(credentialsPath.toFile())
         .appCommandMetricsEnvironment(pluginInfoService.getExternalPluginName())
         .appCommandMetricsEnvironmentVersion(pluginInfoService.getPluginVersion())
-        .appCommandOutputFormat("json")
-        .build();
+        .appCommandOutputFormat("json");
+
+    getProjectJavaSdk(project).ifPresent(sdkBuilder::javaSdkPath);
```

Comment:
@joaoandremartins Yeah this one is a bit marginal.  I think that the jdk location ideally would be surfaced as part of the API for calling the individual commands that require it, so that the project sdk is only read at the time the action is performed.  

However since the scope of a Cloud SDK is usually confined to running a single action, it's something we can fix later if you want.